src/cljs/doit/auth.cljs

```diff
@@ -0,0 +1,52 @@
+(ns doit.auth
+  (:refer-clojure :exclude [subs])
+  (:require [cljs.core.async :refer [put! >! <! take chan buffer ]]
+            [doit.config :as config]
+            [doit.subs :as subs]
+            [doit.events :as events]
+            [re-frame.core :as rf]
+            [goog.object]))
+
+
+(defn get-auth-instance []
+  (.getAuthInstance (goog.object/get js/gapi "auth2")))
+
+(defn save-auth-token
+  [cofx [_ token]]
+  {:db (assoc-in (:db cofx) [:user :token] token)
+   :dispatch [::events/get-todos]})
+
+(rf/reg-event-fx
+ ::save-auth-token
+ save-auth-token)
+
+(defn remove-user-details
+  [cofx _]
+  {:db (dissoc (:db cofx) :user)})
+
+(rf/reg-event-fx
+ ::remove-user-details
+ remove-user-details)
+
+(defn sign-in-success [gapi-user]
+  (let [token (.-id_token (.getAuthResponse gapi-user))]
+    (rf/dispatch [::save-auth-token token])
+    (prn "sign in success")))
+
+(defn sign-in-failure []
+  (prn "sign in failed"))
+
+(defn sign-in []
+  (-> (.signIn (get-auth-instance))
+      (.then sign-in-success sign-in-failure)))
+
+(defn sign-out []
+  (-> (.signOut (get-auth-instance))
+      (.then #(rf/dispatch [::events/initialize-db]))))
```

Comment:
You should instead dispatch `::events/signed-out` so that you can decouple specific events(like `initialize-db`) from `auth`.